Add tests to the layer-openstack

ajkavanagh opened the pull request:
Add unit tests to the adapters, charm and ip parts of the lib to ensure
that the API to the classes remains stable with no unplanned regessions
over time.

ajkavanagh committed:
Add tests to the layer-openstack

Add unit tests to the adapters, charm and ip parts of the lib to ensure
that the API to the classes remains stable with no unplanned regessions
over time.

ajkavanagh commented:
I've tested the build now, and the current charm-tools does respect the ignore in layer.yaml for the layer.  Thus, it should be okay to merge, but you might want to try it yourself first!

thedac commented:
This looks good to me, except, the above message says there is a merge conflict. So that needs to be fixed first.

ajkavanagh commented:
Well it _was_ okay.  I'll go back and rebase it :)

ajkavanagh committed:
Fix dynamic accessor creation on base adapter class

After rebasing on the idiomatic branch it turns out there was an
error on the creation of the dynamic accessors.  This fixes that
issue.

ajkavanagh commented:
This should be okay to go.  I did a rebase to pull in the previous PR and that created all sorts of problems.  I think on PRs we should merge only (which I've always just done in the past).  Anyway, the tests showed a bug in the idiomatic refactor, which I've now fixed as part of this revised PR.
